### PR TITLE
[gulp-task]  Order dependencies

### PR DESCRIPTION
**Pull request by Minh-Van**
[gulp-task]  Order dependencies

### CODE REVIEW

**Comment by Minh-Van** [comment]
NOTICE: I'm removing 2 examples when I merge this branch to develop

**Comment by huuthonguyen76** [comment]
Resolve conflict plz

File: components/example1/example-b.component.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿(function (ngxBootstrap) {
+  ngxBootstrap.ngxComponents.ngxExampleBComponent = ng.core.Directive({
+    selector: 'ngx-example-b',
+  })
+  .Class({
+    constructor: function () {
```
Comment (huuthonguyen76) [inline components/example1/example-b.component.js]: Can we remove this "empty" constructor?

File: components/example1/example1.info.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿{
+  "example1/example-a.component": {
+    "dependencies": [
+      "example1/services/example-a.service",
+      "example1/example-b.component"
+    ]
+  },
+  "example1/example-b.component": { },
```
Comment (huuthonguyen76) [inline components/example1/example1.info.json]: Are those "empty" configs necessary?

File: gulp-tasks/order-dependencies.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+            this.readJsonFile = function (filePath) {
+              if (!_jsonFiles[filePath]) {
+                _jsonFiles[filePath] = JSON.parse(_fileStream.readFileSync(filePath, 'utf8').replace(/^\uFEFF/, ''));
+              }
+
+              return _jsonFiles[filePath];
+            };
+          })();
+          _orderedDependencies = new (function () {
```
Comment (huuthonguyen76) [inline gulp-tasks/order-dependencies.js]: How about:
``` Javascript
_orderedDependencies = {
}
```

File: gulp-tasks/order-dependencies.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+      function _addComponentInfoAndDependencies(key, componentInfo) {
+        if (!componentInfo) { return; }
+
+        if (componentInfo.dependencies) {
+          var _componentKey;
+          for (var i = 0; i < componentInfo.dependencies.length; i++) {
+            _componentKey = componentInfo.dependencies[i];
+
+            if (_orderedDependencies.contain(_componentKey)) { continue; }
```
Comment (huuthonguyen76) [inline gulp-tasks/order-dependencies.js]: Hey i think here we can simplify it by:
```Javascript
if (!_orderedDependencies.contain(_componentKey)) {
  _addComponentInfoAndDependencies(_componentKey, _getComponentInfo(_componentKey));
}
```

File: components/example1/example-b.component.js (hunk 1 above)
Comment (Minh-Van) [inline components/example1/example-b.component.js]: No guy. constructor function is required by angular 2, even if it's a empty function

File: components/example1/example1.info.json (hunk 2 above)
Comment (Minh-Van) [inline components/example1/example1.info.json]: You need declare all components & services at here. If not, gulp task won't know which exactly components/services you want to inject to index.html

File: gulp-tasks/order-dependencies.js (hunk 4 above)
Comment (Minh-Van) [inline gulp-tasks/order-dependencies.js]: Ok, you're right

File: gulp-tasks/order-dependencies.js (hunk 3 above)
Comment (Minh-Van) [inline gulp-tasks/order-dependencies.js]: I need to encapsulate _componentInfos inside of orderedDependencies. So I must use constructor function at here

Comment (huuthonguyen76) [inline gulp-tasks/order-dependencies.js]: Ya, I think this is not "pretty". @thanhtranit, hey what do ya think?

File: components/example1/example-b.component.js (hunk 1 above)
Comment (huuthonguyen76) [inline components/example1/example-b.component.js]: Got it :D

**Pull request closed without merging**